src/colorizer/Dataset.ts

```diff
@@ -625,12 +631,13 @@ export default class Dataset {
   /**
    * Returns the 2D centroid of a given object id.
    */
-  public getCentroid(objectId: number): [number, number] | undefined {
-    const index = objectId * 2;
+  public getCentroid(objectId: number): [number, number, number] | undefined {
+    const index = objectId * 3;
     const x = this.centroids?.[index];
     const y = this.centroids?.[index + 1];
-    if (x && y) {
-      return [x, y];
+    const z = this.centroids?.[index + 2];
+    if (x !== undefined && y !== undefined && z !== undefined) {
+      return [x, y, z];
```

Comment:
nit: with three of these here now, might want to just

```typescript
if (this.centroids === undefined || this.centroids === null) {
  return undefined;
}
```

the `undefined` checks for every individual component might still be good for checking if `objectId` was out of bounds (can't 100% tell whether that was or wasn't the original intent), but I might just check against `this.centroids.length` if so.